diff --git a/hopper/setup.py b/hopper/setup.py
index 7471320..eec861b 100644
--- a/hopper/setup.py
+++ b/hopper/setup.py
@@ -400,44 +400,6 @@ if not SKIP_CUDA_BUILD:
     if bare_metal_version < Version("12.3"):
         raise RuntimeError("FlashAttention-3 is only supported on CUDA 12.3 and above")
 
-    # ptxas 12.8 gives the best perf currently
-    # We want to use the nvcc front end from 12.6 however, since if we use nvcc 12.8
-    # Cutlass 3.8 will expect the new data types in cuda.h from CTK 12.8, which we don't have.
-    if bare_metal_version != Version("12.8"):
-        download_and_copy(
-            name="nvcc",
-            src_func=lambda system, arch, version: f"cuda_nvcc-{system}-{arch}-{version}-archive/bin",
-            dst_path="bin",
-            version=NVIDIA_TOOLCHAIN_VERSION["nvcc"],
-            url_func=lambda system, arch, version:
-            f"https://developer.download.nvidia.com/compute/cuda/redist/cuda_nvcc/{system}-{arch}/cuda_nvcc-{system}-{arch}-{version}-archive.tar.xz",
-        )
-        download_and_copy(
-            name="ptxas",
-            src_func=lambda system, arch, version: f"cuda_nvcc-{system}-{arch}-{version}-archive/bin/ptxas",
-            dst_path="bin",
-            version=NVIDIA_TOOLCHAIN_VERSION["ptxas"],
-            url_func=lambda system, arch, version:
-            f"https://developer.download.nvidia.com/compute/cuda/redist/cuda_nvcc/{system}-{arch}/cuda_nvcc-{system}-{arch}-{version}-archive.tar.xz",
-        )
-        download_and_copy(
-            name="ptxas",
-            src_func=lambda system, arch, version: f"cuda_nvcc-{system}-{arch}-{version}-archive/nvvm/bin",
-            dst_path="nvvm/bin",
-            version=NVIDIA_TOOLCHAIN_VERSION["ptxas"],
-            url_func=lambda system, arch, version:
-            f"https://developer.download.nvidia.com/compute/cuda/redist/cuda_nvcc/{system}-{arch}/cuda_nvcc-{system}-{arch}-{version}-archive.tar.xz",
-        )
-        base_dir = os.path.dirname(__file__)
-        ctk_path_new = os.path.abspath(os.path.join(base_dir, os.pardir, "third_party", "nvidia", "backend", "bin"))
-        nvcc_path_new = os.path.join(ctk_path_new, f"nvcc{exe_extension}")
-        # Need to append to path otherwise nvcc can't find cicc in nvvm/bin/cicc
-        # nvcc 12.8 seems to hard-code looking for cicc in ../nvvm/bin/cicc
-        os.environ["PATH"] = ctk_path_new + os.pathsep + os.environ["PATH"]
-        os.environ["PYTORCH_NVCC"] = nvcc_path_new
-        # Make nvcc executable, sometimes after the copy it loses its permissions
-        os.chmod(nvcc_path_new, os.stat(nvcc_path_new).st_mode | stat.S_IEXEC)
-
     cc_flag = []
     cc_flag.append("-gencode")
     cc_flag.append("arch=compute_90a,code=sm_90a")
