diff --git a/hopper/setup.py b/hopper/setup.py
index 7ed8abc..12e41ad 100644
--- a/hopper/setup.py
+++ b/hopper/setup.py
@@ -401,7 +401,7 @@ if not SKIP_CUDA_BUILD:
     # ptxas 12.8 gives the best perf currently
     # We want to use the nvcc front end from 12.6 however, since if we use nvcc 12.8
     # Cutlass 3.8 will expect the new data types in cuda.h from CTK 12.8, which we don't have.
-    if bare_metal_version != Version("12.8"):
+    if bare_metal_version < Version("12.8"):
         download_and_copy(
             name="nvcc",
             src_func=lambda system, arch, version: f"cuda_nvcc-{system}-{arch}-{version}-archive/bin",
